deploy_vcl

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+
+require 'diffy'
+require 'erb'
+require 'fastly'
+require 'yaml'
+
+require_relative './lib/render_template'
+
+CONFIGS = YAML.load_file(File.join(__dir__, "fastly.yaml"))
+
+def get_config(args)
+  raise "Usage: #{$0} <configuration> <environment>" unless args.size == 2
+
+  configuration = args[0]
+  environment = args[1]
+  config_hash = CONFIGS[configuration][environment] rescue nil
+
+  raise "ERROR: Unknown configuration/environment combination. Check this combination exists in fastly.yaml." unless config_hash
+
+  return configuration, environment, config_hash
+end
+
+def get_git_version
+  ref = %x{git describe --always}.chomp
+  ref = "unknown" if ref.empty?
+
+  ref
+end
+
+def get_dev_version(configuration)
+  # Sometimes the latest version isn't the development version.
+  version = configuration.version
+  version = version.clone if version.active?
+
+  version
+end
+
+def delete_ui_objects(service_id, version_number)
+  # Delete objects created by the UI. We want VCL to be the source of truth.
+  # Most of these don't have real objects in the Fastly API gem.
+  to_delete = %w{backend healthcheck cache_settings request_settings response_object header gzip}
+  to_delete.each do |type|
+    type_path = "/service/#{service_id}/version/#{version_number}/#{type}"
+    @f.client.get(type_path).map{ |i| i["name"] }.each do |name|
+      puts "Deleting #{type}: #{name}"
+      resp = @f.client.delete("#{type_path}/#{ERB::Util.url_encode(name)}")
+      raise 'ERROR: Failed to delete configuration' unless resp
+    end
+  end
+end
+
+def modify_settings(version, ttl)
+  settings = version.settings
+  settings.settings.update({
+    "general.default_host" => "",
+    "general.default_ttl"  => ttl,
+  })
+  settings.save!
+end
+
+def upload_vcl(version, contents)
+  vcl_name = 'main'
+
+  begin
+    version.vcl(vcl_name) && version.delete_vcl(vcl_name)
+  rescue Fastly::Error
```

Comment:
I realise it was like this in https://github.com/alphagov/fastly-configure, but is it not useful to puts the error just for visibility? I don't really know how useful that would be or how often it even occurs.

Reply:
Good spot, I'll definitely look into that.